src/core/classes/prism.ts

```diff
@@ -67,6 +67,7 @@ export default class Prism {
 		if (!this.config.manual) {
 			this.waitFor.push(documentReady());
 
+			this.ready = allSettled(this.waitFor);
```

Comment:
For now, this seems to resolve on the next tick, and we don't consider all other pending promises.

Reply:
Is there a promise in the array when this runs? If it resolves before any other promises get added then it won't ever un-resolve.

Reply:
There is, and I think it's a dummy promise we need so `this.ready` has an initial value:

<img width="388" alt="image" src="https://github.com/user-attachments/assets/f6a622f6-177f-4168-8492-feacb0fee422" />